### PR TITLE
DB-10448: improve restore performance

### PR DESCRIPTION
**Pull request by jyuanca**

### CODE REVIEW

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: hbase_storage/src/main/java/com/splicemachine/hbase/ZkUtils.java [hunk 1]
```diff
@@ -315,10 +317,17 @@ public static long nextSequenceId(String counterNode) throws IOException{
...
 
         RecoverableZooKeeper rzk=ZkUtils.getRecoverableZooKeeper();
-        int maxTries=10;
+        int maxTries=Integer.MAX_VALUE;
         int tries=0;
         while(tries<=maxTries){
             tries++;
+            if (tries > 1) {
+                try {
+                    Thread.sleep(50);
```
Comment (dgomezferro) [inline hbase_storage/src/main/java/com/splicemachine/hbase/ZkUtils.java:326]: If we are sleeping, can you use `PipelineUtils.getPauseTime()` or something similar to do an exponential backoff and add jitter?

File: splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/btree/IndexConglomerate.java [hunk 2]
```diff
@@ -151,16 +155,25 @@ protected void create(boolean isExternal,
...
         try{
 //            ((SpliceTransaction)rawtran).elevate(Bytes.toBytes(Long.toString(containerId)));
+
+            TxnView txnView = null;
+            if (rawtran instanceof SpliceTransaction) {
+                txnView = ((SpliceTransaction) rawtran).getTxnInformation();
+            }
+            else if (rawtran instanceof SpliceTransactionView) {
+                txnView = ((SpliceTransactionView) rawtran).getTxnInformation();
+            }
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/btree/IndexConglomerate.java:165]: Not used, remove

Comment (ascend1) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/btree/IndexConglomerate.java:165]: TxnView does not seem to be used in the following code. Looks like line 169 covers both cases here.

**Review by dgomezferro — CHANGES_REQUESTED** [review]

File: hbase_storage/src/main/java/com/splicemachine/hbase/ZkUtils.java [hunk 3]
```diff
@@ -315,10 +317,17 @@ public static long nextSequenceId(String counterNode) throws IOException{
 		 */
 
         RecoverableZooKeeper rzk=ZkUtils.getRecoverableZooKeeper();
-        int maxTries=10;
+        int maxTries=Integer.MAX_VALUE;
```
Comment (ascend1) [inline hbase_storage/src/main/java/com/splicemachine/hbase/ZkUtils.java:320]: Is this common in zk code? I'm not very familiar with this but loop forever gives a strange feel...

**Review by ascend1 — CHANGES_REQUESTED** [review]

**Review by dgomezferro — APPROVED** [review]

File: splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/btree/IndexConglomerate.java (hunk 2 above)
Comment (jyuanca) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/btree/IndexConglomerate.java]: This has been deleted

File: hbase_storage/src/main/java/com/splicemachine/hbase/ZkUtils.java (hunk 3 above)
Comment (jyuanca) [reply hbase_storage/src/main/java/com/splicemachine/hbase/ZkUtils.java:321]: Retry until successful.

**Comment by jyuanca** [comment]
Please also review ee branch

**Review by ascend1 — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

**Pull request merged**